### DISCUSSION TITLE
New Battery with Pace BMS don´t work with this project

### DISCUSSION THREAD

**Discussion by KemnaMOL** [syssi/esphome-pace-bms#55] [Q&A]
Hello.
I have a new, second battery with Pace BMS. As the first one (with Daly BMS) I will use the data with MQTT.
I installed the D1 Mini with esphome and add your YAML. Than connect with an RS484-TTL_Converter to the BMS (Address 01) and the D1 Mini. 
It have connection to my MQTT, but without data. The BMS is in RS485 Mode: PACE_MODBUS_485.
The debugger shows no real problem:
```
[07:58:55][D][uart_debug:114]: >>> 01:03:00:00:00:04:44:09
[07:58:56][D][uart_debug:114]: >>> 01:03:00:00:00:04:44:09
[07:58:56][D][uart_debug:114]: >>> 01:03:00:00:00:04:44:09
[07:58:56][D][uart_debug:114]: >>> 01:03:00:00:00:04:44:09
[07:58:56][D][uart_debug:114]: >>> 01:03:00:00:00:04:44:09
[07:58:56][D][modbus_controller:037]: Modbus command to device=1 register=0x00 no response received - removed from send queue
```

Can any one help me and have an Idea?

**Comment by syssi** [comment]
Could you provide a photo of your wiring? Which RS485 port do you use? There are different RS485 converter out there. Which RS485 converter do you use? Is flow control required?

**Comment by KemnaMOL** [comment]
Thanks for your help. I use the port near CAN. At the TTL-Converter only RxD is flashing. The converter is with a MAX485. I don´t know anything about flow control. What is it and who can I check this?
![20240926_181853](https://github.com/user-attachments/assets/6b3b6833-a5bf-45ee-a3ab-d67409f08a63)
![20240926_181916](https://github.com/user-attachments/assets/be5ac829-171c-4a37-bbe1-3d2c05b84b6c)

**Reply by syssi** [reply]
Please swap the RX, TX lines. The TX LED must flash periodically.

**Reply by syssi** [reply]
You module type doesn't need flow control.

**Comment by KemnaMOL** [comment]
You don´t sayd at which end I must swap. :-) OK, first try at RS485 was wrong. Than at the ESP, the TX flash.
But it also gives no data.

**Reply by syssi** [reply]
Okay. Let's focus on the RS485 A/B side now. Why do you use the brown wire? Could you check which color is attached to pin2 of your RJ45 plug? It should be green or orange: https://github.com/syssi/esphome-pipsolar/blob/main/images/rj45-colors-t568a-vs-t568.png

You have to connect `Pin 1` (green-white or orange-white) of the RJ45 plug to `B-` and `Pin 2` (green or orange) to `A+`. See https://github.com/syssi/esphome-pace-bms?tab=readme-ov-file#rj45-jack

**Comment by KemnaMOL** [comment]
Yes, today on toilet I find the same mistake. :-)
I count the RJ45 from wrong side. I have changed to orange (measured). Nothing.
I twisted RX/TX... nothing.
I added the Ground, nothing.

**Reply by syssi** [reply]
Do you have a second RS485 converter and ESP? Could we build a test setup which looks like this:

```
first ESP  <--> first RS485 converter <--> second RS485 converter <--> second ESP
```

**Reply by KemnaMOL** [reply]
Yes, I have different Converter ( works with my Daly2MQTT) and a second ESP. I will try it tomorrow.

**Comment by KemnaMOL** [comment]
The other inverter is a model only with MAX485, which needs a third signal for acc. :-(

**Reply by syssi** [reply]
I don't understand what you are doing in detail. What's your idea?

**Comment by KemnaMOL** [comment]
![20240929_103507](https://github.com/user-attachments/assets/6368155d-7ba3-4e5a-99d1-89d10521f372)
I have this adapter as second model. But I read that it needs some more handshake at the DE/RE Pins.
So I must try it with a second pair equal to my first.

**Reply by syssi** [reply]
We can adapt the YAML a bit to do the required flow control for your converter. If you have some spare time we can give it a try!

**Comment by KemnaMOL** [comment]
So, it´s time for a new try. With new parts there is no cahnge. Now I would try first your esp8266-modbus-dump.yaml
It also brings no data. 
I saw that I wired the PINS D6, D7 on my D1 mini, but thats not right with the program. I changed the yaml to GPIO 12,13. Nothing.
There is no picture which Pins are the right ones.
I have my BMS Dips on 0000 and changed the yaml also to modbus address: 0x00. Is that right?
